comp/dogstatsd/listeners/uds_datagram_test.go

```diff
@@ -91,67 +90,38 @@ func TestUDSDatagramReceive(t *testing.T) {
 		telemetryMock, ok := deps.Telemetry.(telemetry.Mock)
 		assert.True(t, ok)
 
-		registry := telemetryMock.GetRegistry()
-		var udsConnectionsMetric []*dto.Metric
-		var udsPacketsMetric []*dto.Metric
-		var udsPacketsBytesMetric []*dto.Metric
-		var readLatencyMetric []*dto.Metric
-
-		metricsFamily, err := registry.Gather()
+		udsConnectionsMetrics, err := telemetryMock.GetCountMetric("dogstatsd", "uds_connections")
+		assert.Nil(t, err)
```

Comment:
You should use `require` instead of `assert`  as `udsConnectionsMetrics` is used later in this test.
Note: `NoError(t, err)` provides a better error message compared to `Nil(t, err)`